### PR TITLE
Hello World notebook 

### PR DESCRIPTION
**Pull request by SagiPolaczek**
Adding an introductory 'hello world' notebook.

### CODE REVIEW

File: fuse_examples/tutorials/hello_world/hello_world_utils.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        x = self.conv2(x)
+        x = F.relu(x)
+        x = F.max_pool2d(x, 2)
+        x = self.dropout1(x)
+        x = torch.flatten(x, 1)
+        x = self.fc1(x)
+        x = F.relu(x)
+        x = self.dropout2(x)
+        x = self.fc2(x)
+        output = F.log_softmax(x, dim=1)
```
Comment (mosheraboh) [inline fuse_examples/tutorials/hello_world/hello_world_utils.py:35]: @alex-golts , is this log_softmax make sense here?

File: fuse_examples/tutorials/hello_world/hello_world.ipynb [hunk 2]
```diff
@@ -0,0 +1,1411 @@
...
+    "Using the Evaluator from the evaluation package of FuseMedML (fuse.eval) which is a standalone library for evaluating ML models that not necessarily trained with FuseMedML.\n",
+    "\n",
+    "More details and examples for the evaluation package can be found [here](https://github.com/IBM/fuse-med-ml/blob/master/fuse/eval/README.md)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
```
Comment (mosheraboh) [inline fuse_examples/tutorials/hello_world/hello_world.ipynb:659]: remove the long output

File: fuse_examples/tutorials/hello_world/hello_world.ipynb [hunk 3]
```diff
@@ -0,0 +1,1411 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### **Train**\n",
+    "Building Fuse's manager and supplying it PyTorch's optimizer and scheduler.\n",
+    "\n",
+    "The manager is the main API while using Fuse - it resposible for the Train and Infer functionallity.\n",
```
Comment (mosheraboh) [inline fuse_examples/tutorials/hello_world/hello_world.ipynb:340]: Let's remove this line (it's not necessarily the main API)

File: fuse_examples/tutorials/hello_world/hello_world.ipynb [hunk 4]
```diff
@@ -0,0 +1,1411 @@
...
+    "    ('auc', MetricAUCROC(pred='model.output.classification', target='data.label', class_names=class_names)),\n",
+    "])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### **Evaluate**\n",
+    "Using the Evaluator from the evaluation package of FuseMedML (fuse.eval) which is a standalone library for evaluating ML models that not necessarily trained with FuseMedML.\n",
```
Comment (mosheraboh) [inline fuse_examples/tutorials/hello_world/hello_world.ipynb:650]: move it to the begging of the evaluation section

**Review by mosheraboh — CHANGES_REQUESTED** [review]
Thanks @SagiPolaczek, looks good!
See my few minor comments inline.
In addition, 
1. let's add a link to this notebook from the main READM file
2. Let's push an empty hello world notebook (without the outputs)
3. After this PR - please explore the option to add unittest to make sure this notebook works.
4. I've added a comment for @alex-golts as well.

File: fuse_examples/tutorials/hello_world/hello_world.ipynb [hunk 5]
```diff
@@ -0,0 +1,1411 @@
...
+   "metadata": {},
+   "source": [
+    "##### **Imports**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
```
[the remark this answers is not in the document]
Comment (SagiPolaczek) [reply fuse_examples/tutorials/hello_world/hello_world.ipynb:71]: Did you mean to clear the output cell as uploading a clean notebook, or to use python's clear_out functionality to clean it dynamically?

File: fuse_examples/tutorials/hello_world/hello_world.ipynb (hunk 3 above)
Comment (SagiPolaczek) [reply fuse_examples/tutorials/hello_world/hello_world.ipynb:340]: Sure. Also removed it from FuseManagerDefault's documentation :)

File: fuse_examples/tutorials/hello_world/hello_world.ipynb (hunk 2 above)
Comment (SagiPolaczek) [reply fuse_examples/tutorials/hello_world/hello_world.ipynb]: I've used io's walkaround to prevent it from printing.. do you have a nittier way?

File: fuse_examples/tutorials/hello_world/hello_world_utils.py (hunk 1 above)
Comment (alex-golts) [reply fuse_examples/tutorials/hello_world/hello_world_utils.py:35]: @mosheraboh @SagiPolaczek I looked into it. I think it's okay that it's there. the logits before log_softmax are basically the same as after, up to a constant. so applying softmax on x, or on x after log_softmax, is the same.
I think applying the log_softmax can have numerical benefits when calculating CE or NLL loss after it, but I guess that in this simple example it probably doesn't matter much

Comment (mosheraboh) [reply fuse_examples/tutorials/hello_world/hello_world_utils.py:35]: @alex-golts CE applies log_softmax and NLL loss.
In that case, you will apply log_softmax twice 
image

Comment (alex-golts) [reply fuse_examples/tutorials/hello_world/hello_world_utils.py:35]: > @alex-golts CE applies log_softmax and NLL loss.
> In that case, you will apply log_softmax twice
> 

yes, but applying it twice doesn't change the result :-)

Comment (mosheraboh) [reply fuse_examples/tutorials/hello_world/hello_world_utils.py:35]: @alex-golts 
Agree. But it's probably less numerically stable.
I'm approving, but if you agree it's less numerically stable, consider removing it.

**Review by mosheraboh — APPROVED** [review]
Thanks a lot @SagiPolaczek !
Go ahead and merge it.
